docs/reference/commands/node-tool.asciidoc

```diff
@@ -53,19 +54,28 @@ way forward that does not risk data loss, but it may be possible to use the
 `elasticsearch-node` tool to construct a new cluster that contains some of the
 data from the failed cluster.
 
-This tool has two modes:
+If a node that previously held data or were master-eligible is repurposed into
```

Comment:
I suggest:

1. moving this to the top of the `Description` section,
2. making new subheadings e.g. `==== Changing the roles of a node` and `==== Recovering data after a disaster` to separate the sections more clearly, and
3. expanding this paragraph into a longer description of the concepts (and/or describing the concepts in `docs/reference/modules/node.asciidoc` and linking to them).

Reply:
About item 3: I think master-eligible node and data node are already described in `docs/reference/modules/node.asciidoc`. I can certainly link to those descriptions. But it sounds like you wanted something more than that, can you elaborate on that?

Reply:
Sure, sorry, I'll clarify. Today the `Node` docs describe the `node.data` and `node.master` settings but do not say anything about how to go about changing them. This makes sense in 6.x because you don't have to do anything particularly special to change them, but in 7.0 this is no longer the case. I think that page needs to mention that you should only really set these settings before starting a node for the first time, and also talk about (or at least link to) the extra steps you need to do if you want to change them later.